File: app/proprietary/src/main/java/stirling/software/proprietary/classification/ClassificationConditions.java
package stirling.software.proprietary.classification;

import stirling.software.proprietary.document.conditions.Condition;
import stirling.software.proprietary.document.conditions.ConditionInput;

/** Identifies conditions that consume a classifier-produced verdict. */
public final class ClassificationConditions {

    private ClassificationConditions() {}

    /**
     * Missing or unrelated inputs do not require classification; validation handles invalid ones.
     */
    public static boolean requiresClassification(Condition condition) {
        return condition != null
                && condition.input() instanceof ConditionInput.DocumentField(var field)
                && field != null
                && field.startsWith("classification.");
    }
}

File: frontend/editor/src/proprietary/data/classificationConditions.ts
import type { Condition, MatchesAnyCondition } from "@app/conditions/types";

/** Creates the classification comparison offered by the policy wizard and pipeline builder. */
export function classificationCondition(
  values: string[] = [],
): MatchesAnyCondition {
  return {
    input: { source: "document", field: "classification.labels" },
    operator: "matches-any",
    values,
  };
}

/** Classification facts require a classifier-produced verdict; unrelated facts do not. */
export function requiresClassification(condition: Condition): boolean {
  return (
    condition.input.source === "document" &&
    condition.input.field.startsWith("classification.")
  );
}
